src/Response/CSVResponse.php

```diff
@@ -0,0 +1,121 @@
+<?php declare(strict_types = 1);
+
+namespace Contributte\Application\Response;
+
+use Nette\Application\IResponse;
+use Nette\Http\IRequest as IHttpRequest;
+use Nette\Http\IResponse as IHttpResponse;
+
+/**
+ * CSV file download response
+ *
+ * @author Pavel Janda <me@paveljanda.com>
+ */
+class CSVResponse implements IResponse
+{
+
+	/** @var string */
+	protected $contentType = 'application/octet-stream';
+
+	/** @var string */
+	protected $delimiter;
+
+	/** @var mixed[] */
+	protected $data;
+
+	/** @var string */
+	protected $outputEncoding;
+
+	/** @var bool */
+	protected $includeBom;
+
+	/** @var string */
+	protected $name;
+
+	/** @var string[] */
+	protected $headers = [
+		'Expires' => '0',
+		'Cache-Control' => 'no-cache',
+		'Pragma' => 'Public',
+	];
+
+	/**
+	 * @param mixed[] $data
+	 */
+	public function __construct(
+		array $data,
+		string $name = 'export.csv',
+		string $outputEncoding = 'utf-8',
+		// may be often windows-1250 on windows machines
+		string $delimiter = ';',
+		bool $includeBom = FALSE
+	)
+	{
+		if (strpos($name, '.csv') === FALSE) {
+			$name = sprintf('%.csv', $name);
```

Comment:
Maybe `$name = sprintf('%s.csv', $name);`?

Reply:
Yep, it's a bug.

Reply:
Fixed.